packages/eslint-config-udemy-website/index.js

```diff
@@ -23,7 +23,22 @@ module.exports = {
         },
     },
     rules: {
-        'filenames/match-regex': ['error', '^(?:[a-z0-9\\-]+(?:\\.(?:ng-(?:constant|controller|directive|factory|filter|provider|service)|videojs-component|react-(?:component|proptypes)|mobx-(?:model|store)))?(?:\\.spec)?)$'],
+        'filenames/match-regex': [
+            'error',
+            [
+                [
+                    '^(?:_?[a-z0-9\\-]+', // allows underscore start and any lowercase filename
+                    '(?:\\.(?:', // any dot must be:
+                    'ng-(?:constant|controller|directive|factory|filter|provider|service)', // an Angular file
+                    '|videojs-component', // or a videojs-compenent
+                    '|react-(?:isocomponent|component|proptypes)', // or a React file
+                    '|mobx-(?:model|store)))?', // or a Mobx file
```

Comment:
Maybe move the last `))?` to a new line to match with `'(?:\\.(?:', // any dot must be:` on its own line.

Reply:
I'll do that. We could also get really fancy and make all of the groupings their own array. ;-)